### DISCUSSION TITLE
How can I get the new value after a user drags?

### DISCUSSION THREAD

**Discussion by haijialiu** [microsoft/microsoft-ui-xaml#9501] [Q&A] [answered]
I noticed the Slider of WPF have Thumb.DragCompleted event to get new value, but it seems that can't in WinUI. So, What can I do? Thanks a million.

**Comment by castorix** [comment]
You can use [ValueChanged](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.slider?view=windows-app-sdk-1.5#examples)
or Bind the Value (like in this test [MainWindow.xaml](https://github.com/castorix/WinUI3_SwapChainPanel_Direct2D/blob/master/MainWindow.xaml) with _GetSpeed_ and _SetSpeed_ functions)

**Reply by haijialiu** [reply]
It seems like get value on time (that means if drags, I will get to many values ), but I want get the value after drag.

**Comment by castorix** [comment, marked as answer]
> It seems like get value on time (that means if drags, I will get to many values ), but I want get the value after drag.

For the event when you release the Thumb, you can call _ManipulationCompleted_
In XAML : 
`ManipulationMode="All" ManipulationCompleted="slider1_ManipulationCompleted"`
and
```
        private void slider1_ManipulationCompleted(object sender, ManipulationCompletedRoutedEventArgs e)
        {
           var value = ((Slider)sender).Value;
           // ...
        }
```

**Reply by haijialiu** [reply]
That's the answer I've always wanted. Thank you so much.
